plugins/inputs/snmp/snmp.go

```diff
@@ -149,6 +131,48 @@ func (s *Snmp) init() error {
 	return nil
 }
 
+func (s *Snmp) getMibsPath() {
+	gosmi.Init()
+	var folders []string
+	for _, mibPath := range s.Path {
+		gosmi.AppendPath(mibPath)
+		folders = append(folders, mibPath)
+		err := filepath.Walk(mibPath, func(path string, info os.FileInfo, err error) error {
+			// symlinks are files so we need to double check if any of them are folders
+			// Will check file vs directory later on
+			if info.Mode()&os.ModeSymlink != 0 {
+				link, err := os.Readlink(path)
+				if err != nil {
+					s.Log.Warnf("Bad symbolic link %v")
+				}
+				folders = append(folders, link)
+			}
+			return nil
+		})
+		if err != nil {
+			s.Log.Warnf("Filepath could not be walked %v", err)
```

Comment:
After thinking about it forth and back I concluded that it might be best to return an error here and thus make `init` fail as we probably have some bigger issue if we cannot walk the given path. What do you think?

Reply:
My thought behind this is that some filepaths could be walked but one might fail, ex I have 3 file paths but only one can't be walked I would like to walk what I have but know what couldn't be. So overall telegraf won't stop if it can still process other mibs.

Reply:
This has been changed into an error. Thank you so much for your input.